### DISCUSSION TITLE
Running Incus VM's on WSL-2

### DISCUSSION THREAD

**Discussion by mwoodpatrick** [microsoft/WSL#11167] [Q&A]
Has anyone gotten Incus VM's to work on WSL-2. I'm able to launch Incus containers without issue but when I run:

`incus launch images:ubuntu/22.04 ubuntu-vm --vm`

I get:

```
Launching ubuntu-vm
Error: Failed instance creation: Failed creating instance record: Instance type "virtual-machine" is not supported on this server: vhost_vsock kernel module not loaded
```

**Comment by hi-ko** [comment]
I think it's a more general issue that wsl2's kernel is compiled without `CONFIG_VHOST_VSOCK` kernel module included:

```bash
sudo modprobe vhost_vsock
modprobe: FATAL: Module vhost_vsock not found in directory /lib/modules/5.15.153.1-microsoft-standard-WSL2
```

similar issue with zfs kernel module: [Module zfs not found in /lib/modules/5.15.90 #10128](https://github.com/microsoft/WSL/issues/10128)
